assembly/src/assembler/mod.rs

```diff
@@ -130,8 +137,20 @@ impl Assembler {
         let source = source.as_ref();
         let program = ProgramAst::parse(source)?;
 
+        // fill up the imported procedures map
+        let mut import_info_vec = Vec::new();
+        for module in self.module_provider.module_iter() {
+            if let Some(import_info) = module.ast().import_info() {
+                import_info_vec.push(import_info);
+            }
+        }
```

Comment:
I think this creates quite a bit of extra work, and in most cases this work will probably not be needed. Instead, I think we should be able to build this map on per-module basis. For example, we could pass `ModuleAst` to `Context::being_module()` function (line 213 below) and then build this map for the passed-in module inside that function.